File: app/src/main/java/net/chris/demo/guardian/model/inject/ListModule.java
package net.chris.demo.guardian.model.inject;

import net.chris.demo.guardian.ui.model.ListViewModel;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ListModule {

    @Provides
    @Singleton
    protected ListViewModel provideListViewModel() {
        return new ListViewModel();
    }

}

File: app/src/main/kotlin/net/chris/demo/guardian/model/inject/ListModule.kt
package net.chris.demo.guardian.model.inject

import dagger.Module
import dagger.Provides
import net.chris.demo.guardian.ui.model.ListViewModel
import javax.inject.Singleton

@Module
class ListModule {

    @Provides
    @Singleton
    fun provideListViewModel(): ListViewModel = ListViewModel()

}
